python/taichi/lang/ast/ast_transformer.py

```diff
@@ -508,6 +509,52 @@ def warn_if_is_external_func(ctx, node):
             node.lineno + ctx.lineno_offset,
             module="taichi")
 
+    @staticmethod
+    # extract format specifier from raw_string, and also handles positional arguments, if there are any
+    def extract_format_spec(raw_string, *raw_args, **raw_keywords):
+        raw_brackets = re.findall(r'{(.*?)}', raw_string)
```

Comment:
Let's add some comments to explain the meaning of `raw_args` and `raw_keywords` with an example. For example (if I understand it correctly):
```
# original input: ti.format("zxcxc{.3f}", x: ti.Expr)
# raw_string: "zxczxc{.3f}"
# raw_args: x (ti.Expr)
# raw_keywords: ...
```